src/app/rundown-view/components/offsetable-piece/offsetable-piece.component.ts

```diff
@@ -33,6 +36,51 @@ export class OffsetablePieceComponent {
   @ViewChild('labelTextElement')
   public labelTextElement: ElementRef<HTMLSpanElement>
 
+  public media: Media | undefined
+
+  public isMediaPiece: boolean = false
+
+  private mediaSubscription?: Subscription
+
+  constructor(
+    private readonly mediaStateService: MediaStateService,
+    private readonly changeDetectorRef: ChangeDetectorRef
+  ) {}
+
+  public updatePieceMedia(): void {
+    if (!this.doesPieceContainMedia()) {
+      this.isMediaPiece = false
+      return
+    }
+    const mediaSourceName: string = this.getPieceMediaSourceName(this.piece)
+    if (!mediaSourceName) {
+      this.isMediaPiece = false
+      return
+    }
+    this.isMediaPiece = true
+    this.mediaSubscription = this.mediaStateService.subscribeToMedia(mediaSourceName).subscribe(this.updateMediaAvailabilityStatus.bind(this))
+  }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    const pieceChange: SimpleChange | undefined = changes['piece']
+    if (pieceChange) {
+      const oldMediaSourceName: string | undefined = pieceChange.previousValue ? this.getPieceMediaSourceName(pieceChange.previousValue) : undefined
+      if (oldMediaSourceName === this.getPieceMediaSourceName(pieceChange.currentValue)) {
+        return
+      }
+      this.mediaSubscription?.unsubscribe()
+      this.updatePieceMedia()
+    }
+  }
+
+  private doesPieceContainMedia(): boolean {
+    const piece: Tv2Piece = this.piece as Tv2Piece
+    if (!piece.metadata.sourceName) {
+      return false
+    }
+    return !!piece.metadata.sourceName
```

Comment:
I haven't checked, but `return !!piece.metadata.sourceName` should be enough.

Reply:
It was enough!